### PR TITLE
Fixes #1031: (7) Added support for ARRAYSIZE to PROPERTY.ARRAY in tupleparser; test cases

### LINKED ISSUE #1031: array size on property arrays is ignored in tupleparser
In `parse_property_array()`, ArraySize is tolerated but ignored.

### PR DESCRIPTION
**Pull request by andy-maier**
**Note:** This PR is on top of PR #1029.
For details, see the commit message (of only the last commit).
Ready for review and merge.

### CODE REVIEW

File: pywbem/tupletree.py [hunk 1]
```diff
@@ -41,9 +41,9 @@
 
 The NAME is the name of the element.
 
-The ATTRS are a name-value hash of element attributes.
+The ATTRS are a name-value dictionary of element attributes, not pres. order.
```
Comment (KSchopmeyer) [inline pywbem/tupletree.py]: what does pres. mean

Comment (KSchopmeyer) [reply pywbem/tupletree.py]: ahha.  preserving.  Lets spell it out.

File: pywbem/tupleparse.py [hunk 2]
```diff
@@ -1201,7 +1232,7 @@ def parse_property(tup_tree):
                 'EMBEDDEDOBJECT'],
                ['QUALIFIER', 'VALUE'])
 
-    quals = {}
+    quals = OrderedDict()
```
Comment (KSchopmeyer) [inline pywbem/tupleparse.py]: Why OrderedDict rather than NocaseDict???

Comment (andy-maier) [reply pywbem/tupleparse.py]: The dict of the input arg is not used anyway; its items are put into a NocaseDict by the implementation of CIMProperty, so OrderedDict was the slightly cheaper alternative.

However, looking at the code again after your comment, it occurred to me that we now can pass in the list returned from list_matching() directly. Like we already do for CIMClass.

I'll change that.

File: pywbem/tupletree.py (hunk 1 above)
Comment (andy-maier) [reply pywbem/tupletree.py]: Was too long for the line. But you are right, we should spell it out.

File: pywbem/tupleparse.py (hunk 2 above)
Comment (andy-maier) [reply pywbem/tupleparse.py]: DONE

File: pywbem/tupletree.py (hunk 1 above)
Comment (andy-maier) [reply pywbem/tupletree.py]: DONE

**Pull request merged**